packages/@aws-cdk/aws-lambda-nodejs/lib/bundling.ts

```diff
@@ -67,7 +72,7 @@ export class Bundling implements cdk.BundlingOptions {
 
     Bundling.esbuildInstallation = Bundling.esbuildInstallation ?? EsbuildInstallation.detect();
 
-    this.projectRoot = path.dirname(props.depsLockFilePath);
+    this.projectRoot = props.projectRoot ?? path.dirname(props.depsLockFilePath);
```

Comment:
This copy operation

https://github.com/aws/aws-cdk/blob/5f597877c75f9e92d3bf08eedb5007ecc3cb001e/packages/%40aws-cdk/aws-lambda-nodejs/lib/bundling.ts#L153

will not work anymore if the lockfile is not directly under `projectRoot`.

Reply:
I'm not sure this logic can be separate from the assumption that the lockfile is directly in the `projectRoot`

Thoughts on throwing an error if `props.nodeModules` is used when the lockfile is in a different location?

Reply:
I think we can determine the path of the lock file relative to the project root and use it in the copy operation when we join with `inputDir`?

Reply:
Added. @jogold let me know what you think